core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -713,15 +735,24 @@ public void updateTask(TaskResult taskResult) {
         }
 
         String workflowId = taskResult.getWorkflowInstanceId();
-        WorkflowModel workflowInstance = executionDAOFacade.getWorkflowModel(workflowId, false);
-
-        TaskModel task =
-                Optional.ofNullable(executionDAOFacade.getTaskModel(taskResult.getTaskId()))
-                        .orElseThrow(
-                                () ->
-                                        new NotFoundException(
-                                                "No such task found by id: %s",
-                                                taskResult.getTaskId()));
+        String shardId = (String) taskResult.getOutputData().get("shardId");
+        WorkflowModel workflowInstance;
+        TaskModel task;
+        LOGGER.info("Inside updateTask - shardId - {}", shardId);
+        if (StringUtils.isNotEmpty(shardId)) {
```

Comment:
- This could be a separate method returning Optional<Task>
- Post the method call we can do orElseThrow

Reply:
done